Android examples

jrmerwin opened the pull request:
## What changes were proposed in this pull request?

(Please fill in changes proposed in this fix)

## How was this patch tested?

(Please explain how this patch was tested. E.g. unit tests, integration tests, manual tests)

Please review
https://github.com/deeplearning4j/deeplearning4j/blob/master/CONTRIBUTING.md before opening a pull request.

crockpotveggies commented:
@saudet if you could take a quick look and give your blessing I'll get this merged.

saudet commented on android/DL4JImageRecognitionDemo/app/build.gradle:
```diff
@@ -0,0 +1,70 @@
+        testCompile 'junit:junit:4.12'
+
+        //Image loading dependencies
+        compile 'org.datavec:datavec-data-codec:0.9.1'
```
We don't need `datavec-data-codec` here, please use `datavec-data-image` instead to reduce size.

Also, binaries of OpenCV, Leptonica, and HDF5 for other platforms are what is taking most of the space here. We should add `exclude` lines for `opencv-platform`, `leptonica-platform`, and `hdf5-platform`, and add back maybe just the binaries for OpenCV on Android.

And please add these instructions (datavec-data-image and platform artifact exclusion) to the docs at https://github.com/deeplearning4j/deeplearning4j-docs/pull/7/. Thanks!

jrmerwin replied:
I changed the image loading dependency to datavec-data-image, excluded leptonica and hdf5, and left opencv. The tutorial documents and demo code have been updated.

saudet commented on android/DL4JImageRecognitionDemo/app/src/androidTest/java/com/example/jmerwin/dl4jimagerecognitiondemo/ExampleInstrumentedTest.java:
```diff
@@ -0,0 +1,26 @@
+        // Context of the app under test.
+        Context appContext = InstrumentationRegistry.getTargetContext();
+
+        assertEquals("com.example.jmerwin.dl4jimagerecognitiondemo", appContext.getPackageName());
```
Rename package to org.deeplearning4j.examples!

saudet commented on android/DL4JIrisClassifierDemo/DL4JIrisClassifierDemo Tutorial.md:
```diff
@@ -0,0 +1,272 @@
+        compile 'org.nd4j:nd4j-native:0.9.1'
+        compile 'org.nd4j:nd4j-native:0.9.1:android-x86'
+        compile 'org.nd4j:nd4j-native:0.9.1:android-arm'
+        compile 'org.bytedeco.javacpp-presets:systems-platform:1.4'
```
Update dependencies, excludes, here and below as well, but this looks like a copy/paste of the docs?

jrmerwin replied:
I forgot there were copies of the tutorials in the demo folders. If it is OK with you I will just remove them since there is a link to the tutorial in the readme. Then I won't have to keep two copies up to date.

saudet replied:
Yes, sounds good, let's just keep the links. Thanks!

jrmerwin commented on android/DL4JImageRecognitionDemo/README.md:
```diff
@@ -0,0 +1,8 @@
+Using Deeplearning4J in Android Applications
+
+This tutorial demonstrates how the Java-based deep learning library Deeplearning4J can use a trained neural network in Android Applications. The application loads a neural network trained on the Mnist data set and uses it to identify user drawn numbers as input. The key features of this demo include storing and accessing a trained model stored in the application’s Raw folder, internally saving an ...
+A tutorial for the application can be found [here](https://github.com/jrmerwin/Skymind-Android-Documentation/blob/master/DL4JImageRecognitionDemo.md#head_link6).
```
updated

jrmerwin commented:
Changed the package names in both demo apps, excluded opencv in the ImageDemo followed by include statements for android specific opencvs. Updated the demo tutorial with the new exclusions.

saudet commented:
Thanks! Looks good. Just one last thing: The README.md file is empty. It might make sense to just merge the 2 ones found in the project subdirectories? Anyway, it's a really minor point. I'm good with merging this for now and let's get this updated for 1.0.0-alpha first. :)
